turbinia/workers/analysis/wordpress_creds.py

```diff
@@ -0,0 +1,192 @@
+        result.close(self, success=True, status='No Wordpress database found')
+        return result
+    except TurbiniaException as e:
+      result.close(
+          self, success=True,
+          status='Error retrieving Wordpress database: {0:s}'.format(str(e)))
+      return result
+
+    try:
+      (creds, hashnames) = self._extract_wordpress_hashes(location)
+    except TurbiniaException as e:
+      result.close(self, success=True, status=str(e))
+      return result
+
+    (report, priority, summary) = self._analyse_wordpress_creds(
+        creds, hashnames)
+    output_evidence.text_data = report
+    result.report_data = report
+    result.report_priority = priority
+
+    # Write the report to the output file.
+    with open(output_file_path, 'wb') as fh:
+      fh.write(output_evidence.text_data.encode('utf8'))
+      fh.write('\n'.encode('utf8'))
+
+    # Add the resulting evidence to the result object.
+    result.add_evidence(output_evidence, evidence.config)
+    result.close(self, success=True, status=summary)
+
+    return result
+
+  def _collect_wordpress_file(self, evidence):
+    """Extract artifacts using image_export.
+
+    Args:
+        evidence (Evidence object):  The evidence to process
+    Returns:
+        location (str): The file path to the extracted evidence.
+        number of artifacts (int): The number of files extracted.
+    """
+    try:
+      collected_artifacts = extract_files(
+          file_name=_WP_DB_NAME, disk_path=evidence.local_path,
+          output_dir=os.path.join(self.output_dir, 'artifacts'))
+    except TurbiniaException as e:
+      raise TurbiniaException('artifact extraction failed: {}'.format(str(e)))
+
+    # Extract base dir from our list of collected artifacts
+    location = os.path.dirname(collected_artifacts[0])
+
+    return (location, len(collected_artifacts))
+
+  def _extract_wordpress_hashes(self, location):
+    """Dump the Wordpress credentials from a raw database file.
+
+    Args:
+        location (list): Directory of files extracted from the disk.
+
+    Returns:
+        creds (list): List of strings containing raw extracted credentials
+        hashnames (dict): Dict mapping hash back to username for convenience.
+
+    Raises:
+      TurbiniaException: when the process fails.
+    """
+    creds = []
+    hashnames = {}
+    for dirs, _, files in os.walk(location):
+      if _WP_DB_NAME in files:
+        try:
+          strings = subprocess.Popen(
+              ['strings', os.path.join(dirs, _WP_DB_NAME)],
+              stdout=subprocess.PIPE, text=True)
+          grep = subprocess.run(['grep', r'\$P\$'], stdin=strings.stdout,
+                                check=False, text=True, capture_output=True)
+          if grep.returncode == 1:
+            raise TurbiniaException('No Wordpress credentials found')
+          if grep.returncode == 2:
+            raise TurbiniaException(
+                'Error grepping file: {}'.format(grep.stdout))
```

Comment:
super minor nit, but can we use format strings similar to `{0:s}` for consistency with the style guide?

Reply:
Done